packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1049,64 +1010,159 @@ export class KubernetesClient {
+  }
+
+  /**
+   * Apply a given yaml file to a context, i.e. 'kubectl apply -f'. Resources that exist
+   * on the context are patched, and any new resources are created.
+   *
+   * @param context a context
+   * @param filePath file system path to a YAML Kubernetes spec
+   * @param namespace the namespace to use for any resources that don't include one
+   * @return an array of resources created
+   */
+  async applyResourcesFromFile(context: string, filePath: string, namespace?: string): Promise<KubernetesObject[]> {
+    const manifests = await this.loadManifestsFromFile(filePath);
+    return this.syncResources(context, manifests, 'apply', namespace);
+  }
+
+  /**
+   * Apply a given yaml file to a context, i.e. 'kubectl apply -f'. Resources that exist
+   * on the context are patched, and any new resources are created.
+   *
+   * @param context a context
+   * @param filePath file system path to a YAML Kubernetes spec
+   * @param namespace the namespace to use for any resources that don't include one
+   * @return an array of resources created
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async applyResources(context: string, manifests: any[], namespace?: string): Promise<KubernetesObject[]> {
+    return this.syncResources(context, manifests, 'apply', namespace);
+  }
+
+  /**
+   * Applies or creates a set of resources to a context, via creation or patching.
+   *
+   * @param context a context
+   * @param manifests a set of Kubernetes spec manifests
+   * @param action 'create' (only create new resources, do not overwrite existing) or 'apply' (create or patch as necessary)
+   * @param namespace the namespace to use for any resources that don't include one
+   * @return an array of resources created
+   *
+   * Heavily influenced by the API example:
+   * https://github.com/kubernetes-client/javascript/blob/0fbfd8fc2dcc7f4ec3e6fcd64a5c55169b6ef0b8/examples/typescript/apply/apply-example.ts
+   */
+
+  async syncResources(
+    context: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    manifests: any[],
+    action: 'create' | 'apply',
+    namespace?: string,
+  ): Promise<KubernetesObject[]> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const telemetryOptions: Record<string, any> = {
+      manifestsSize: manifests?.length,
+      action: action,
+    };
+    if (namespace) {
+      telemetryOptions.namespace = namespace;
+    }
+
     try {
       const ctx = new KubeConfig();
       ctx.loadFromFile(this.kubeconfigPath);
       ctx.currentContext = context;
-      for (const manifest of manifests) {
-        // https://github.com/kubernetes-client/javascript/issues/487
-        if (manifest?.metadata?.creationTimestamp) {
-          manifest.metadata.creationTimestamp = new Date(manifest.metadata.creationTimestamp);
+
+      const validSpecs = manifests.filter(s => s?.kind);
+
```

Comment:
it looks like we had a fix before
```
        // https://github.com/kubernetes-client/javascript/issues/487
        if (manifest?.metadata?.creationTimestamp) {
          manifest.metadata.creationTimestamp = new Date(manifest.metadata.creationTimestamp);
```

that is no longer present in the new method.
The ticket being still opened I suppose we still need the fix on manifests ?

Reply:
by finding when we added the code either we can have a test case probably

Reply:
The problem here was due to type conversion between the Kubernetes Typescript API (timestamp is Date | undefined) vs Kubernetes REST API (date-time string | undefined). In the old path functions like createNamespacedDeploymen() were parsing the JSON into Kube API objects locally and then serializing them to send the data, so this conversion was necessary. With the new path we're doing a lower-level patch/create which AFAICT isn't trying to convert (and the apply example wasn't doing this), so shouldn't be necessary.

Reply:
ok 👍